Tests/UnitTests/Core/TrackFitting/Gx2fTests.cpp

```diff
@@ -74,33 +74,43 @@ static std::vector<Acts::SourceLink> prepareSourceLinks(
   return result;
 }
 
+/// @brief Create a simple telescope detector
+///
+/// @param geoCtx
+/// @param nSurfaces Number of surfaces
 std::shared_ptr<const TrackingGeometry> makeToyDetector(
-    const GeometryContext& tgContext, const std::size_t nSurfaces = 5) {
+    const Acts::GeometryContext& geoCtx, const std::size_t nSurfaces = 5) {
   if (nSurfaces < 1) {
     throw std::invalid_argument("At least 1 surfaces needs to be created.");
   }
+
+  // Define the dimensions of the square surfaces
+  const double halfSizeSurface = 1_m;
+
+  // Rotation of the surfaces
+  const double rotationAngle = M_PI * 0.5;
+  const Vector3 xPos(cos(rotationAngle), 0., sin(rotationAngle));
+  const Vector3 yPos(0., 1., 0.);
+  const Vector3 zPos(-sin(rotationAngle), 0., cos(rotationAngle));
+
   // Construct builder
   CuboidVolumeBuilder cvb;
 
   // Create configurations for surfaces
   std::vector<CuboidVolumeBuilder::SurfaceConfig> surfaceConfig;
-  for (unsigned int i = 1; i <= nSurfaces; i++) {
+  for (unsigned int surfPos = 1; surfPos <= nSurfaces; surfPos++) {
```

Comment:
And shouldn't it be `std::size_t`?
```suggestion
  for (std::size_t surfInd = 1; surfInd <= nSurfaces; surfInd++) {
```

Reply:
I think both `surfInd` and `surfPos` can be valid, but I decided to go for position, since those are also the positions (in m) where we put the surfaces. `surfInd` might be a bit misleading, since the surface gets a different index than that one assigned.

`unsigned int` over `size_t` because we only want positive numbers here.